sdcm/nemesis.py

```diff
@@ -3412,7 +3412,7 @@ def start_and_interrupt_rebuild_streaming(self):
         trigger = partial(
             self.target_node.run_nodetool, sub_cmd="rebuild", warning_event_on_exception=(Exception,), retry=0,
         )
-        log_follower = self.target_node.follow_system_log(patterns=["Rebuild starts"])
+        log_follower = self.target_node.follow_system_log(patterns=["rebuild from dc:"])
```

Comment:
It must be versioned.
Either by using decorator like at [1] or version in place version comparison [2].

[1] https://github.com/scylladb/scylla-cluster-tests/blob/f5b13d5a/sdcm/nemesis.py#L2855
[2] https://github.com/scylladb/scylla-cluster-tests/blob/d28c419969d6e8b5835fb8b9d77f95808ca21cfe/sdcm/cluster.py#L2739

Reply:
IIUC what @asias  said, we have `rebuild from dc` also for version where RBNO is not enabled by default...
do we need to keep it to much older versions?

Reply:
> IIUC what @asias said, we have `rebuild from dc` also for version where RBNO is not enabled by default...

That's correct
Here is the log with old rebuild:
```
2023-03-02T11:31:12+00:00 gemini-1tb-10h-fix-byo--db-node-428011cc-4     !INFO | scylla[13608]:  [shard  0] api - rebuild: source_dc=
2023-03-02T11:31:12+00:00 gemini-1tb-10h-fix-byo--db-node-428011cc-4     !INFO | scylla[13608]:  [shard  0] storage_service - rebuild from dc: (any dc)
2023-03-02T11:31:12+00:00 gemini-1tb-10h-fix-byo--db-node-428011cc-4     !INFO | scylla[13608]:  [shard  0] storage_service - enable_repair_based_node_ops=true, allowed_repair_based_node_ops={replace}
2023-03-02T11:31:12+00:00 gemini-1tb-10h-fix-byo--db-node-428011cc-4     !INFO | scylla[13608]:  [shard  0] range_streamer - Rebuild starts, nr_ranges_remaining=3597
2023-03-02T11:31:12+00:00 gemini-1tb-10h-fix-byo--db-node-428011cc-4     !INFO | scylla[13608]:  [shard  0] range_streamer - Rebuild with [10.12.3.251, 10.12.3.128] for keyspace=system_distributed started, nodes_to_stream=2
```
and here is the one with repair-based rebuild
```
2023-03-03T09:14:04+00:00 gemini-1tb-10h-master-db-node-4467dd96-4     !INFO | scylla[5113]:  [shard  0] api - rebuild: source_dc=
2023-03-03T09:14:04+00:00 gemini-1tb-10h-master-db-node-4467dd96-4     !INFO | scylla[5113]:  [shard  0] storage_service - rebuild from dc: (any dc)
2023-03-03T09:14:04+00:00 gemini-1tb-10h-master-db-node-4467dd96-4     !INFO | scylla[5113]:  [shard  0] storage_service - enable_repair_based_node_ops=true, allowed_repair_based_node_ops={decommission, bootstrap, rebuild, removenode, replace}
2023-03-03T09:14:04+00:00 gemini-1tb-10h-master-db-node-4467dd96-4     !INFO | scylla[5113]:  [shard  0] repair - rebuild_with_repair: started with keyspaces=system_tracessystem_authks1system_distributed_everywheresystem_distributed, source_dc=us-east, nr_ranges_total=11833, ignore_nodes=[]
```
Both have `rebuild from dc:` message

Reply:
What are those `much older versions`? It is too ambigiuos statement.
If it is `4.x` then ok. But, if it is `5.x` then it makes sense to make versioning.

Reply:
we are not supporting any 4.x versions anymore, so i think if the "old" version here is `5.0`, we are good to go

Reply:
In emails, I found the logs from the 81st build of `longevity-10gb-3h-test` on scylla-4.5 (test id: `f41aa0d9-8410-4edf-88c1-8f675223f406`, scylla version: `4.5.7-0.20220618.f0a846534 (ami-033972dfb05ee2e69)`). The link to Jenkins build is already dead of course.

The target node of `RebuildStreamingErr` is `longevity-10gb-3h-4-5-db-node-f41aa0d9-4`. It has the following in the log:
```
2022-06-20T20:23:43+00:00  longevity-10gb-3h-4-5-db-node-f41aa0d9-4 !INFO    | scylla:  [shard 0] storage_service - rebuild from dc: (any dc)
2022-06-20T20:23:44+00:00  longevity-10gb-3h-4-5-db-node-f41aa0d9-4 !INFO    | scylla:  [shard 0] range_streamer - Rebuild starts, nr_ranges_remaining=3515
2022-06-20T20:23:44+00:00  longevity-10gb-3h-4-5-db-node-f41aa0d9-4 !INFO    | scylla:  [shard 0] range_streamer - Rebuild with [10.0.0.118, 10.0.2.20, 10.0.1.54, 10.0.1.7, 10.0.2.125] for keyspace=keyspace1 started, nodes_to_stream=5
```

For 5.0 I looked at the logs from this build https://jenkins.scylladb.com/job/scylla-5.0/job/longevity/job/longevity-100gb-4h-test/49/
The target node is `longevity-100gb-4h-5-0-db-node-b7e060f7-3`
```
2023-02-23T23:12:40+00:00 longevity-100gb-4h-5-0-db-node-b7e060f7-3 !    INFO |  [shard 0] api - rebuild: source_dc=
2023-02-23T23:12:40+00:00 longevity-100gb-4h-5-0-db-node-b7e060f7-3 !    INFO |  [shard 0] storage_service - rebuild from dc: (any dc)
2023-02-23T23:12:40+00:00 longevity-100gb-4h-5-0-db-node-b7e060f7-3 !    INFO |  [shard 0] storage_service - enable_repair_based_node_ops=true, allowed_repair_based_node_ops={replace}
2023-02-23T23:12:41+00:00 longevity-100gb-4h-5-0-db-node-b7e060f7-3 !    INFO |  [shard 2] compaction - [Compact keyspace1.standard1 9273faf0-b3cf-11ed-b67d-cd90e3a16273] Compacting [/var/lib/scylla/data/keyspace1/standard1-596b9740b3b111eda48a3e394297c4c2/md-8290-big-Data.db:level=0:origin=memtable,/var/lib/scylla/data/keyspace1/standard1-596b9740b3b111eda48a3e394297c4c2/md-5420-big-Data.db:level=0:origin=memtable]
2023-02-23T23:12:41+00:00 longevity-100gb-4h-5-0-db-node-b7e060f7-3 !    INFO |  [shard 13] compaction - [Compact keyspace1.standard1 927f1e80-b3cf-11ed-b5ec-cd99e3a16273] Compacting [/var/lib/scylla/data/keyspace1/standard1-596b9740b3b111eda48a3e394297c4c2/md-8287-big-Data.db:level=0:origin=memtable,/var/lib/scylla/data/keyspace1/standard1-596b9740b3b111eda48a3e394297c4c2/md-8259-big-Data.db:level=0:origin=memtable]
2023-02-23T23:12:42+00:00 longevity-100gb-4h-5-0-db-node-b7e060f7-3 !    INFO |  [shard 4] compaction - [Compact keyspace1.standard1 92c52830-b3cf-11ed-9007-cd8fe3a16273] Compacting [/var/lib/scylla/data/keyspace1/standard1-596b9740b3b111eda48a3e394297c4c2/md-6332-big-Data.db:level=0:origin=memtable,/var/lib/scylla/data/keyspace1/standard1-596b9740b3b111eda48a3e394297c4c2/md-8278-big-Data.db:level=0:origin=memtable]
2023-02-23T23:12:42+00:00 longevity-100gb-4h-5-0-db-node-b7e060f7-3 !    INFO |  [shard 1] compaction - [Compact keyspace1.standard1 92ce01d0-b3cf-11ed-be21-cd92e3a16273] Compacting [/var/lib/scylla/data/keyspace1/standard1-596b9740b3b111eda48a3e394297c4c2/md-8289-big-Data.db:level=0:origin=memtable,/var/lib/scylla/data/keyspace1/standard1-596b9740b3b111eda48a3e394297c4c2/md-5279-big-Data.db:level=0:origin=memtable]
2023-02-23T23:12:42+00:00 longevity-100gb-4h-5-0-db-node-b7e060f7-3 !    INFO |  [shard 5] compaction - [Compact keyspace1.standard1 930d06a0-b3cf-11ed-9f7e-cd91e3a16273] Compacting [/var/lib/scylla/data/keyspace1/standard1-596b9740b3b111eda48a3e394297c4c2/md-5115-big-Data.db:level=0:origin=memtable,/var/lib/scylla/data/keyspace1/standard1-596b9740b3b111eda48a3e394297c4c2/md-8279-big-Data.db:level=0:origin=memtable]
2023-02-23T23:12:42+00:00 longevity-100gb-4h-5-0-db-node-b7e060f7-3 !    INFO |  [shard 0] range_streamer - Rebuild starts, nr_ranges_remaining=20396
2023-02-23T23:12:42+00:00 longevity-100gb-4h-5-0-db-node-b7e060f7-3 !    INFO |  [shard 0] range_streamer - Rebuild with [10.0.1.71, 10.0.1.146, 10.0.3.28, 10.0.3.199, 10.0.0.228] for keyspace=drop_table_during_repair_ks_3 started, nodes_to_stream=5
```

`rebuild from dc:` is present everywhere

Reply:
thank you @ilya-rarov

Reply:
@ilya-rarov that addresses my concern. Thanks. No versioning is needed then.